### PR TITLE
[14.0] [MIG] l10n_ch_invoice_reports: Migrate to version 14.0

### PR DESCRIPTION
**Pull request by sonhd91**

### CODE REVIEW

File: l10n_ch_invoice_reports/data/reports.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        <field name="model">account.move</field>
+        <field name="report_type">qweb-pdf</field>
+        <field
+            name="report_name"
+        >l10n_ch_invoice_reports.invoice_isr_report_main</field>
+        <field
+            name="report_file"
+        >l10n_ch_invoice_reports.invoice_isr_report_main</field>
+        <field name="print_report_name">'Invoice-ISR-%s' % object.name</field>
+        <field name="attachment_use">True</field>
```
Comment (sebalix) [inline l10n_ch_invoice_reports/data/reports.xml]: The previous value was `attachment_use="False"`, so:
```suggestion
        <field name="attachment_use" eval="False"/>
```

File: l10n_ch_invoice_reports/data/reports.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    </record>
+
+    <record id="account_move_qr_report" model="ir.actions.report">
+        <field name="name">Invoice with ISR</field>
+        <field name="model">account.move</field>
+        <field name="report_type">qweb-pdf</field>
+        <field name="report_name">l10n_ch_invoice_reports.invoice_qr_report_main</field>
+        <field name="report_file">l10n_ch_invoice_reports.invoice_qr_report_main</field>
+        <field name="print_report_name">'Invoice-QR-%s' % object.name</field>
+        <field name="attachment_use">False</field>
```
Comment (sebalix) [inline l10n_ch_invoice_reports/data/reports.xml]:
```suggestion
        <field name="attachment_use" eval="False"/>
```

File: l10n_ch_invoice_reports/data/reports.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        <field name="model">account.move</field>
+        <field name="report_type">qweb-pdf</field>
+        <field
+            name="report_name"
+        >l10n_ch_invoice_reports.invoice_qr_isr_report_main</field>
+        <field
+            name="report_file"
+        >l10n_ch_invoice_reports.invoice_qr_isr_report_main</field>
+        <field name="print_report_name">'Invoice-QR-ISR-%s' % object.name</field>
+        <field name="attachment_use">False</field>
```
Comment (sebalix) [inline l10n_ch_invoice_reports/data/reports.xml]:
```suggestion
        <field name="attachment_use" eval="False"/>
```

File: l10n_ch_invoice_reports/models/report.py [hunk 4]
```diff
@@ -37,24 +37,32 @@ def merge_pdf_in_memory(self, docs):
...
         if self.report_name not in reports or not res_ids:
-            return super().render_qweb_pdf(res_ids, data)
+            return super()._render_qweb_pdf(res_ids, data)
+
+        if not self.env["account.move"].browse(res_ids).partner_bank_id:
+            record = self.env["account.move"].browse(res_ids)
+            record._compute_bank_partner_id()
+            record.partner_bank_id = (
+                record.bank_partner_id.bank_ids and record.bank_partner_id.bank_ids[0]
+            )
```
Comment (sebalix) [inline l10n_ch_invoice_reports/models/report.py]: I'm not sure to understand this piece of code? Have you some info about this?
I don't think having this kind of data updates in the method `_render_qweb_pdf` is a good idea.

**Review by sebalix — CHANGES_REQUESTED** [review]

File: l10n_ch_invoice_reports/data/reports.xml (hunk 1 above)
Comment (sonhd91) [reply l10n_ch_invoice_reports/data/reports.xml]: Thanks for review, the branch was updated

File: l10n_ch_invoice_reports/data/reports.xml (hunk 2 above)
Comment (sonhd91) [reply l10n_ch_invoice_reports/data/reports.xml]: Thanks for review, the branch was updated

File: l10n_ch_invoice_reports/data/reports.xml (hunk 3 above)
Comment (sonhd91) [reply l10n_ch_invoice_reports/data/reports.xml]: Thanks for review, the branch was updated

File: l10n_ch_invoice_reports/models/report.py (hunk 4 above)
Comment (sonhd91) [reply l10n_ch_invoice_reports/models/report.py]: Thanks for review, I'm looking into it

**Review by mmequignon — APPROVED** [review]

**Review by TDu — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

**Pull request merged**